kernel/comps/virtio/src/device/network/config.rs

```diff
@@ -78,10 +78,17 @@ impl VirtioNetConfig {
     }
 
     pub(super) fn read(this: &SafePtr<Self, IoMem>) -> ostd::prelude::Result<Self> {
+        let mut mac_data: [u8; 6] = [0; 6];
+        let mut mac_ptr = field_ptr!(this, Self, mac);
+        for byte in &mut mac_data {
+            *byte = mac_ptr.clone().cast::<u8>().read_once().unwrap();
+            mac_ptr.byte_add(1);
+        }
+
         Ok(Self {
             // FIXME: It is impossible to call `read_once` on `EthernetAddr`. What's the proper way
             // to read this field out?
```

Comment:
Could you please remove this `FIXME`?

However, I think we should put another TODO above. Something like:
```rust
        // TODO: Add a general API to read this byte-by-byte.
        let mut mac_data: [u8; 6] = [0; 6];
        let mut mac_ptr = field_ptr!(this, Self, mac);
        for byte in &mut mac_data {
            *byte = mac_ptr.clone().cast::<u8>().read_once().unwrap();
            mac_ptr.byte_add(1);
        }
```
as well as
```rust
        // TODO: Add a general API to read this byte-by-byte.
        let mut data_ptr = field_ptr!(&self.config, VirtioInputConfig, data);
        let mut out = Vec::with_capacity(size);
        for _ in 0..size {
            out.push(data_ptr.clone().cast::<u8>().read_once().unwrap());
            data_ptr.byte_add(1);
        }
```

Also, what prevents us from adding the general API in this PR? Choosing a method name?

Reply:
The PR was originally intended to use the existing API to solve the reading problem under TDX. I think we can consider using the general API later.